python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Error Labeling Manager class."""
+
+from copy import deepcopy
+from enum import Enum
+
+import numpy as np
+import torchvision
+from torch import Tensor
+
+LABELS = 'labels'
+
+
+class ErrorLabelType(Enum):
+    """
+    Enum providing types of error labels.
+
+    If none, then the detection is not an error. It is a
+    correct prediction.
+    """
+
+    # the gt doesn't have a corresponding detection
+    MISSING = "missing"
+
+    # the model predicted detections, but there was nothing there
+    # this prediction must have a 0 iou score with all gt detections
+    BACKGROUND = "background"
+
+    # the predicted class is correct, bounding box is not
+    LOCALIZATION = "localization"
+
+    # the predicted class is incorrect, the bounding box is correct
+    CLASS_NAME = "class_name"
+
+    # both the predicted class and bounding box are incorrect
+    CLASS_LOCALIZATION = "class_localization"
+
+    # the predicted class is correct, the bounding box is correct, but
+    # the iou score is lower than another detection
+    DUPLICATE_DETECTION = "duplicate_detection"
+
+    MATCH = "match"
+
+
+class ErrorLabeling():
+    """Defines a wrapper class of Error Labeling for vision scenario.
+
+    Only supported for object detection at this point.
+    """
+
+    def __init__(self,
+                 task_type: str,
+                 pred_y: str,
+                 true_y: str,
+                 iou_threshold: float = 0.5):
+        """Create an ErrorLabeling object.
+
+        :param model: The model to explain.
```

Comment:
Perhaps you mean an ml-wrapper wrapped model here (which implements the predict & predict proba methods right?), perhaps can just namedrop that

Reply:
Oops - I didn't mean to reference a model at all! Looks like that was an incorrect copy/paste. Will correct